x-pack/plugin/core/src/main/java/org/elasticsearch/license/internal/TrialLicenseEra.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.license.internal;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.xcontent.ToXContentFragment;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Sometimes we release a version with a bunch of cool new features, and we want people to be able to start a new trial license in a cluster
+ * that's already used a trial and let it expire. This class controls when we do that. The serialization of this class is designed to
+ * maintain compatibility with old-school Elasticsearch versions (specifically the {@link org.elasticsearch.Version} class).
+ */
+public class TrialLicenseEra implements ToXContentFragment, Writeable {
```

Comment:
nit: (feel free ignore) I would prefer `TrialLicenseVersion` over `TrialLicenseEra` custom concept scoped versions will become (if not already) common place and I think `Era` will be the outlier w.r.t. naming.

Reply:
Hah, I used `Era` instead of `Version` to try to avoid arguments about introducing yet another Version class, but maybe those have already been had. I'm very open to calling it `Version`.

Reply:
+1 to using the Version suffix, it's all the rage